web/src/typeahead_helper.ts

```diff
@@ -8,28 +8,35 @@ import * as buddy_data from "./buddy_data";
 import * as compose_state from "./compose_state";
 import {page_params} from "./page_params";
 import * as people from "./people";
+import type {PseudoMentionUser, User} from "./people";
 import * as pm_conversations from "./pm_conversations";
 import * as pygments_data from "./pygments_data";
 import * as recent_senders from "./recent_senders";
 import * as stream_data from "./stream_data";
 import * as stream_list_sort from "./stream_list_sort";
+import type {StreamSubscription} from "./sub_store";
 import * as user_groups from "./user_groups";
+import type {UserGroup} from "./user_groups";
 import * as user_status from "./user_status";
+import type {UserStatusEmojiInfo} from "./user_status";
 import * as util from "./util";
+
+export type UserOrMention = PseudoMentionUser | (User & {is_broadcast: undefined});
```

Comment:
@andersk I'm curious how you feel about this sort of thing. It's definitely easier than going through all the callsites and adding things like `{type: "user"}` or `{type: "mention_user"}`. 

I imagine as we convert more files to Typescript we'd have to change to that, but it would be easier to ensure the callsites are doing the right thing if they're also Typescript, and this works fine for now.

I'm also happy to go through the callsites now if that's the solution you think is important now. I just wanted to check before doing that work.

And I have the same question for `UserGroup | (User & {members: undefined})`

Reply:
This seems fine to me at least -- seems like it is a localized TODO that we can follow-up on in parallel with converting some of the code paths that import this file (most of which wouldn't use this type anyway).